src/sushi.sol

```diff
@@ -96,9 +139,51 @@ contract SushiJoin is CropJoin {
         }
         super.flee();
     }
+    function cage() external {
+        require(live, "SushiJoin/not-live");
 
-    function cage() external auth {
+        // Allow caging if any assumptions change
+        require(
+            wards[msg.sender] == 1 ||
+            masterchef.migrator() != initialMigrator ||
+            masterchef.owner() != address(timelockOwner)
+        , "SushiJoin/not-authorized");
+
+        _cage();
+    }
+    function cage(address target, uint256 value, string memory signature, bytes memory data, uint256 eta) external {
+        require(live, "SushiJoin/not-live");
+
+        // Verify the queued transaction is targetting one of the dangerous functions on Masterchef
+        bytes memory callData;
+        if (bytes(signature).length == 0) {
+            callData = data;
+        } else {
+            callData = abi.encodePacked(bytes4(keccak256(bytes(signature))), data);
+        }
+        require(callData.length >= 4, "SushiJoin/invalid-calldata");
+        bytes4 selector = bytes4(
+            (uint32(uint8(callData[0])) << 24) |
+            (uint32(uint8(callData[1])) << 16) |
+            (uint32(uint8(callData[2])) << 8) |
+            (uint32(uint8(callData[3])))
+        );
+        require(target == address(masterchef), "SushiJoin/wrong-target");
```

Comment:
Why even have `target` as an argument? Seems like you can remove the argument and this check, and just use `masterchef` to get `txHash` below.

Reply:
Good call.